code/campaignclassic/src/main/java/com/adobe/marketing/mobile/CarouselPushTemplate.java

```diff
@@ -0,0 +1,91 @@
+    // Required, "default" or "filmstrip"
+    private String carouselLayoutType;
+    // Required, One or more Items in the carousel defined by the CarouselItem class
+    private ArrayList<CarouselItem> carouselItems = new ArrayList<>();
+
+    class CarouselItem {
+        // Required, URI to an image to be shown for the carousel item
+        private final String imageUri;
+        // Optional, caption to show when the carousel item is visible
+        private final String captionText;
+        // Optional, URI to handle when the item is touched by the user. If no uri is provided for the item, adb_uri will be handled instead.
+        private final String interactionUri;
+        CarouselItem(final String imageUri, final String captionText, final String interactionUri) {
+            this.imageUri = imageUri;
+            this.captionText = captionText;
+            this.interactionUri = interactionUri;
+        }
+
+        public String getImageUri() {
+            return imageUri;
+        }
+
+        public String getCaptionText() {
+            return captionText;
+        }
+
+        public String getInteractionUri() {
+            return interactionUri;
+        }
+    }
+
+    public String getCarouselOperationMode() {
+        return carouselOperationMode;
+    }
+
+    public String getCarouselLayoutType() {
+        return carouselLayoutType;
+    }
+
+    public ArrayList<CarouselPushTemplate.CarouselItem> getCarouselItems() {
+        return carouselItems;
+    }
+
+    CarouselPushTemplate(@NonNull final Map<String, String> messageData) {
+        super(messageData);
+        this.carouselOperationMode = DataReader.optString(messageData, CampaignPushConstants.PushPayloadKeys.CAROUSEL_OPERATION_MODE, "");
+        try {
+            this.carouselLayoutType = DataReader.getString(messageData, CampaignPushConstants.PushPayloadKeys.CAROUSEL_LAYOUT);
+            final List<Map<String, String>> carouselItemMaps = DataReader.getTypedListOfMap(String.class, messageData, CampaignPushConstants.PushPayloadKeys.CAROUSEL_ITEMS);
+            for (final Map<String, String> carouselItemMap : carouselItemMaps) {
+                final String carouselImage = carouselItemMap.get(CampaignPushConstants.PushPayloadKeys.CAROUSEL_ITEM_IMAGE);
+                if (StringUtils.isNullOrEmpty(carouselImage)) {
+                    break;
+                }
+                final String text = carouselItemMap.get(CampaignPushConstants.PushPayloadKeys.CAROUSEL_ITEM_TEXT);
+                final String uri = carouselItemMap.get(CampaignPushConstants.PushPayloadKeys.CAROUSEL_ITEM_URI);
+                final CarouselItem carouselItem = new CarouselItem(carouselImage, text, uri);
+                carouselItems.add(carouselItem);
+            }
+        } catch (final DataReaderException dataReaderException) {
+            Log.debug(CampaignPushConstants.LOG_TAG, SELF_TAG, "Required fields \"adb_car_layout\" or \"adb_items\" not found.");
+        }
```

Comment:
wrapping the the entire logic in a try block will lead this construction to fail even if nullable elements are absent.
Can you use `DataReader.get` only for required values and use `DataReader.opt` for the rest. Also reccommend splitting try catch for required values separately.